packages/salesforcedx-vscode-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -75,16 +101,172 @@ export class CheckpointService implements TreeDataProvider<BaseNode> {
+      uriInput,
+      lineInput
+    );
+    if (cpNode) {
+      const index = this.checkpoints.indexOf(cpNode, 0);
+      if (index > -1) {
+        this.checkpoints.splice(index, 1);
+        this._onDidChangeTreeData.fire();
+        if (makeCommandCall === true) {
+          await this.executeRemoveApexExecutionOverlayActionCommand(cpNode);
+        }
+        return true;
+      }
+    }
+    return false;
+  }
+
+  // recreateIfDupe defaults to true, when the recurisve call is made
+  // this is set to false to prevent
+  private async executeCreateApexExecutionOverlayActionCommand(
+    theNode: CheckpointNode,
+    recreateIfDupe = true
+  ) {
+    // create the overlay action
+    const overlayActionCommand = new ApexExecutionOverlayActionCommand(
+      theNode.createJSonStringForOverlayAction()
+    );
+    let errorString = undefined;
+    let returnString = undefined;
+    await this.myRequestService.execute(overlayActionCommand).then(
+      value => {
+        returnString = value;
+      },
+      reason => {
+        errorString = reason;
+      }
+    );
+    // The resturn string will be the overlay Id and will end up being
+    // used if the node is deleted
+    if (returnString) {
+      const result = JSON.parse(
+        returnString
+      ) as ApexExecutionOverlaySuccessResult;
+      theNode.setActionCommandResult(result.id, undefined);
+    }
+    // Failure could mean that there is already an entry with the same the criteria.
+    // If that's the case then passe the Id out, delete it and recreate it.
+    // Fun fact: the result is an array of 1 item
+    // "[{"message":"duplicate value found: ScopeId duplicates value on record with id: 1doxx00000000Pn","errorCode":"DUPLICATE_VALUE","fields":[]}]"
+    if (errorString) {
+      const result = <ApexExecutionOverlayFailureResult[]>JSON.parse(
+        errorString
+      );
+      // If the node is a dupe, then delete it and recreate it. If the failure reason
+      // was something else that we don't handle then just set the command result.
+      if (result[0].errorCode === DUPLICATE_VALUE && recreateIfDupe) {
+        // parse the record ID from the result[0].message, call remove and then call add again
+        const duplicateId = result[0].message
+          .substr(result[0].message.lastIndexOf(':') + 1)
+          .trim();
+        // set the id to the duplicate id and call the delete to nuke the record,
+        // then call the delete again with the recreation flag unset
+        theNode.setActionCommandResult(duplicateId, undefined);
+        await this.executeRemoveApexExecutionOverlayActionCommand(theNode);
+        // clear out the action command result
+        theNode.setActionCommandResult(undefined, undefined);
+        await this.executeCreateApexExecutionOverlayActionCommand(
+          theNode,
+          false /* recreateIfDupe */
+        );
+      } else {
+        theNode.setActionCommandResult(undefined, result[0].message);
```

Comment:
This experience is different than typical breakpoints. In VS Code, if a breakpoint is not valid, the color turns gray. Looking at this, if it fails to create a checkpoint on the server side, the conditional breakpoint and checkpoint on the client side still exist but will an error message. The conditional breakpoint is probably still red.

What can we do to make the experience more holistic? Should we show an error message to the user and delete the breakpoint & checkpoint? Should we show an error message to the user, delete the checkpoint, and mark the breakpoint as invalid?